### DISCUSSION TITLE
Third party UI

### DISCUSSION THREAD

**Discussion by waghanza** [the-benchmarker/web-frameworks#4165] [Ideas]
Hi,

I can easily output a json like that @SuspiciousLookingOwl (missing language and framework version but I can add them)

```json
[
  {"id":2,"language":"go","framework":"fasthttp","level":64,"label":"average_latency","value":672}
   {"id":2,"language":"go","framework":"fasthttp","level":64,"label":"duration_ms","value":15046734}
   {"id":2,"language":"go","framework":"fasthttp","level":64,"label":"http_errors","value":0}
   {"id":2,"language":"go","framework":"fasthttp","level":64,"label":"maximum_latency","value":47634.333333333336}
   {"id":2,"language":"go","framework":"fasthttp","level":64,"label":"minimim_latency","value":26.333333333333332}
   {"id":2,"language":"go","framework":"fasthttp","level":64,"label":"percentile_50","value":262}
   {"id":2,"language":"go","framework":"fasthttp","level":64,"label":"percentile_75","value":625}
   {"id":2,"language":"go","framework":"fasthttp","level":64,"label":"percentile_90","value":1293}
   {"id":2,"language":"go","framework":"fasthttp","level":64,"label":"percentile_99","value":7738}
   {"id":2,"language":"go","framework":"fasthttp","level":64,"label":"percentile_99.999","value":42092}
   {"id":2,"language":"go","framework":"fasthttp","level":64,"label":"request_timeouts","value":0}
   {"id":2,"language":"go","framework":"fasthttp","level":64,"label":"socket_connection_errors","value":0}
   {"id":2,"language":"go","framework":"fasthttp","level":64,"label":"socket_read_errors","value":0}
   {"id":2,"language":"go","framework":"fasthttp","level":64,"label":"socket_write_errors","value":0}
   {"id":2,"language":"go","framework":"fasthttp","level":64,"label":"standard_deviation","value":1564.3333333333333}
   {"id":2,"language":"go","framework":"fasthttp","level":64,"label":"total_bytes_received","value":279808805}
   {"id":2,"language":"go","framework":"fasthttp","level":64,"label":"total_requests","value":2616385.6666666665}
   {"id":2,"language":"go","framework":"fasthttp","level":64,"label":"total_requests_per_s","value":173883.66666666666}
]
```

**Comment by SuspiciousLookingOwl** [comment]
This is great! More info = better 👌

**Reply by waghanza** [reply]
fyi : those a raw output of https://github.com/wg/wrk

**Comment by waghanza** [comment]
@SuspiciousLookingOwl I think something like will be accurate

```json
{
  "metrics": [
    {
      "level": 512,
      "label": "duration_ms",
      "value": 15098809.333333334,
      "framework_id": 2
    }
  ],
  "frameworks": [
    {
      "id": 2,
      "version": 1.23,
      "language": "go",
      "website": "pkg.go.dev/github.com/valyala/fasthttp"
    }
  ],
  "languages": [
    {
      "label": "go",
      "version": 1.16
    }
  ]
}
```

could be available on next run ❤️

**Reply by SuspiciousLookingOwl** [reply]
I was thinking something like
```json
[
  {
    "id": 1,
    "language": {
        "name": "go",
        "version": "1.16"
    },
    "framework": {
        "name": "fasthttp",
        "version": "1.11"
    },
    "level64": {
        "average_latency": 1,
        "duration_ms": 1,
        "http_errors": 1,
        "maximum_latency": 1,
        ...
    },
    "level256": {
        "average_latency": 1,
        "duration_ms": 1,
        "http_errors": 1,
        "maximum_latency": 1,
        ...
    },
    "level512": {
        "average_latency": 1,
        "duration_ms": 1,
        "http_errors": 1,
        "maximum_latency": 1
        ...
    }
  },
  {
    "id": 2,
    "language": {
        "name": "go",
        "version": "1.16"
    },
    "framework": {
        "name": "express",
        "version": "5."
    },
    "level64": { ... },
    "level256": { ... },
    "level512": { ... }
  },
  ...
]
```
because it's more readable, but there will be a lot of redundant data. Yours seems good too, the data can be easily parsed on the client side.

**Reply by waghanza** [reply]
yeah. the main point is to have a set that you could use :-) 

I'm aware the format is not very proper, but the more important is to a allow parsing from a json stream :-)

**Comment by SuspiciousLookingOwl** [comment]
Btw is it possible to include star count and last commit / update date of each framework's repository, or is it our of your / this repo's scope? 😁

I feel like being able to filter out less popular / less maintained framework is a great feature to add.

**Reply by waghanza** [reply]
> is it our of your / this repo's scope?

I think so. This repository is more about exposing some insight collect on run (performance for example).

Exposing some subjective data is for UI.

> I feel like being able to filter out less popular / less maintained framework is a great feature to add.

Not sure about that, but feel free to open a separate discussion for this subject

**Comment by SuspiciousLookingOwl** [comment]
Hi @waghanza!

I've added the home page as you requested. I also added metric selector for the data table, and changed the layout slightly. Let me know what you think.

https://web-frameworks-benchmark.netlify.app

**Reply by waghanza** [reply]
Yeah ... Let transfer you app in this organisation then (I've sent you an invitation)

**Reply by SuspiciousLookingOwl** [reply]
I've joined to organization, however I need a permission to create a public repository to transfer ownership.

**Reply by waghanza** [reply]
I am stuck playing with github's organization rights. I prefer to give you only write rights to a specific repository, for now

**Reply by SuspiciousLookingOwl** [reply]
No worries. I have a question since I'm still unsure on what to do.

Do I use the [transfer repository feature](https://docs.github.com/en/github/administering-a-repository/transferring-a-repository), or do I use the existing repo ([frontend](https://github.com/the-benchmarker/frontend)) and overwrite there?

I never used the transfer repository feature before but I assume it will create a new repo under the-benchmarker organization (the-benchmarker/web-frameworks-benchmark)

**Reply by waghanza** [reply]
I've gave you maintainer role, so as you can transfer it (better to transfer in order to keep git history).

I prefer a better solution, but better not to block you (I'll search later).

Let me know if any issue on transfer

**Reply by SuspiciousLookingOwl** [reply]
I still can't seem to transfer the repository 👀 

![image](https://user-images.githubusercontent.com/32597776/115975964-6cdb0280-a593-11eb-8f87-7bcdcc1b44f7.png)

**Reply by waghanza** [reply]
Can you know ?

**Reply by SuspiciousLookingOwl** [reply]
Yep, here's the repo: [the-benchmarker/web-frameworks-benchmark](https://github.com/the-benchmarker/web-frameworks-benchmark)
